### PR TITLE
Feat/scala3 hermant

### PR DESCRIPTION
**Pull request by tomasherman**
attempt at scala3 support based on `feat/scala3`.

What is it about:
- we had to change config derivation for scala3 ... it's not source compatible with scala2 so i needed to create specific code for all pureconfig modules for scala-2 and scala-3 codes. See docs here: https://pureconfig.github.io/docs/scala-3-derivation.html

Whats missing:
- [x] rewrite few tests for scala3 that are now just for scala2
- [x] docs about pureconfig and scala3

What had to change:
- in `check` task we now only check `mdoc` task for scala3 (we used to do it for all scala versions) - i don't know how to check it for all considering source code is now diverging for scala2 and scala3
- few scalafix options had to be removed as they are scala2 only. I believe it's ok because scala3 is now the 'main' dialect and therefore we apply scalafix just for it, other scala versions are just xcompiled
- scala3 pureconfig modules no longer support camel case - that is a limitation of pureconfig, we just reflect the change (see https://pureconfig.github.io/docs/scala-3-derivation.html)

### CODE REVIEW

**Comment by tomasherman** [comment]
wohoo finally green

File: project/BuildSettings.scala
Comment (sideeffffect) [inline project/BuildSettings.scala:34]: I'm not sure about having Scala 3 as the default is a good idea.
Wouldn't having 2.13 as the default save us a lot of headache with Scalafix (among other things)?

File: project/BuildSettings.scala
Comment (sideeffffect) [inline project/BuildSettings.scala:40]: TODO :wink:

**Review by sideeffffect — APPROVED** [review]
Looks great, thanks for putting in the effort. But aren't we too hard on ourselves with Scala 3 _as the default_?

File: project/BuildSettings.scala
Comment (tomasherman) [reply project/BuildSettings.scala:40]: good catch

File: project/BuildSettings.scala
Comment (tomasherman) [reply project/BuildSettings.scala:34]: idk, in my view scala3 is the future so i would vote for shaping the code towards that ... i would also like to move further towards scala3 code in future PRs using compiler toggles etc

**Comment by tomasherman** [comment]
(marked as draft so that it's not merged until everything is in place)

**Comment by tomasherman** [comment]
ok i think this is ready for review if anyone else is interested

File: project/BuildSettings.scala
Comment (jakubjanecek) [reply project/BuildSettings.scala:34]: My first choice would probably also be `2.13` for now but if everything works and we can compile to both versions the default is probably not that important.

File: project/Dependencies.scala
Comment (jakubjanecek) [inline project/Dependencies.scala:20]:
```suggestion
  val jetbrainsAnnotations = "org.jetbrains" % "annotations" % "21.0.1"
```

**Review by jakubjanecek — APPROVED** [review]
Thank you!

File: project/BuildSettings.scala
Comment (tomasherman) [reply project/BuildSettings.scala:34]: it only makes a big difference for mdoc imho ... if its all the same to you, i would try to coarse the code into scala3 compatible code as much as possible so i think scala3 default is the best way to go

**Comment by tomasherman** [comment]
Solved all the comments, im going to merge this once all checks pass. Next steps for me is to massage the code into more scala 3 style of code. I will leave it up to @jakubjanecek to release this once he feels like it

**Comment by tomasherman** [comment]
thanks for taking the time to review

**Pull request merged**
